src/controllers/TwitterController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Twitter-specific endpoints live here
+ */
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Subscriber\Oauth\Oauth1;
+
+class TwitterController extends ApiController {
+    public function handle(Request $request, $db) {
+        // really need to not require this to be declared
+    }
+
+    public function getRequestToken($request, $db){
+        // only trusted clients can change account details
+        $clientId = $request->getParameter('client_id');
+        $clientSecret = $request->getParameter('client_secret');
+        $this->oauthModel = $request->getOauthModel($db);
+        if (!$this->oauthModel->isClientPermittedPasswordGrant($clientId, $clientSecret)) {
+            throw new Exception("This client cannot perform this action", 403);
+        }
+
+        // API call to twitter, get request token
+        $client = new Client([
+            'base_url' => 'https://api.twitter.com/',
+            'defaults' => ['auth' => 'oauth']
+
+        ]);
+
+        $oauth = new Oauth1([
+            'consumer_key'    => $this->config['twitter']['consumer_key'],
+            'consumer_secret' => $this->config['twitter']['consumer_secret'],
+        ]);
+        $client->getEmitter()->attach($oauth);
+
+        $res = $client->post('oauth/request_token');
+        if($res->getStatusCode() == 200) {
+            parse_str($res->getBody(), $data);
+
+            $requestTokenMapper = new TwitterRequestTokenMapper($db);
+            // $tokens is instance of TwitterRequestTokenModelCollection
+            $tokens = $requestTokenMapper->create($data['oauth_token'], $data['oauth_token_secret']);
+            $output_list = $tokens->getOutputView($request);
+            header("Location: " . $output_list['twitter_request_tokens'][0]['uri'], NULL, 201);
+            return $output_list;
+        } else {
+
+            error_log("Twitter: no request token (" . $res->getStatusCode() . ": " . $res->getBody() . ")");
+            exit;
```

Comment:
The client gets a 200 with a blank body in this situation. We need to throw an exception with an appropriate status code.